AsyncDisplayKit/ASImageNode.mm

```diff
@@ -153,88 +147,104 @@ - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer;
 {
   BOOL hasValidCropBounds = _cropEnabled && !CGRectIsNull(_cropDisplayBounds) && !CGRectIsEmpty(_cropDisplayBounds);
 
-  return [[_ASImageNodeDrawParameters alloc] initWithCrop:_cropEnabled
-                                                   opaque:self.opaque
-                                                    image:self.image
-                                                   bounds:(hasValidCropBounds ? _cropDisplayBounds : self.bounds)
-                                            contentsScale:self.contentsScaleForDisplay
-                                          backgroundColor:self.backgroundColor
-                                              contentMode:self.contentMode
-                                                 cropRect:self.cropRect
-                                   imageModificationBlock:self.imageModificationBlock];
+  return [[_ASImageNodeDrawParameters alloc] initWithBounds:self.bounds
+                                                     opaque:self.opaque
+                                              contentsScale:self.contentsScaleForDisplay
+                                            backgroundColor:self.backgroundColor
+                                                contentMode:self.contentMode];
 }
 
-+ (UIImage *)displayWithParameters:(_ASImageNodeDrawParameters *)parameters isCancelled:(asdisplaynode_iscancelled_block_t)isCancelled
+- (UIImage *)displayWithParameters:(_ASImageNodeDrawParameters *)parameters isCancelled:(asdisplaynode_iscancelled_block_t)isCancelled
 {
-  UIImage *image = parameters.image;
-
+  ASDN::MutexLocker l(_imageLock);
```

Comment:
If at all possible, we should minimize the critical section to a compact set of lines that accesses all of the needed information to put it into local variables from the instance variables, then unlocks and allows the rest of the method to proceed